database/engine/metadata_log/metadatalogapi.c

```diff
@@ -0,0 +1,290 @@
+}
+
+void metalog_commit_delete_chart(RRDSET *st)
+{
+    struct metalog_instance *ctx;
+    BUFFER *buffer;
+    RRDHOST *host = st->rrdhost;
+
+    /* Metadata are only available with dbengine */
+    if (!host->rrdeng_ctx || RRD_MEMORY_MODE_DBENGINE != st->rrd_memory_mode)
+        return;
+
+    ctx = host->rrdeng_ctx->metalog_ctx;
+    buffer = buffer_create(64); /* This will be freed after it has been committed to the metadata log buffer */
+
+    buffer_sprintf(buffer, "CONTEXT %s\n", host->machine_guid);
+
+    buffer_sprintf(buffer, "TOMBSTONE %s\n", st->id); /* TODO: replace this with real GUID when available */
+
+    metalog_commit_creation_record(ctx, buffer);
+}
+
+void metalog_commit_update_dimension(RRDDIM *rd)
+{
+    struct metalog_instance *ctx;
+    BUFFER *buffer;
+    RRDSET *st = rd->rrdset;
+    RRDHOST *host = st->rrdhost;
+    char uuid_str[37];
+
+    /* Metadata are only available with dbengine */
+    if (!host->rrdeng_ctx || RRD_MEMORY_MODE_DBENGINE != st->rrd_memory_mode)
+        return;
+
+    ctx = host->rrdeng_ctx->metalog_ctx;
+    buffer = buffer_create(128); /* This will be freed after it has been committed to the metadata log buffer */
+
+    buffer_sprintf(buffer, "CONTEXT %s\n", st->id);
+
+    uuid_unparse_lower(*rd->state->rrdeng_uuid, uuid_str);
+    buffer_sprintf(buffer, "GUID %s\n", uuid_str);
+
+    buffer_sprintf(
+        buffer
+        , "DIMENSION \"%s\" \"%s\" \"%s\" " COLLECTED_NUMBER_FORMAT " " COLLECTED_NUMBER_FORMAT " \"%s %s %s\"\n"
+        , rd->id
+        , rd->name
+        , rrd_algorithm_name(rd->algorithm)
+        , rd->multiplier
+        , rd->divisor
+        , rrddim_flag_check(rd, RRDDIM_FLAG_OBSOLETE)?"obsolete":""
+        , rrddim_flag_check(rd, RRDDIM_FLAG_HIDDEN)?"hidden":""
+        , rrddim_flag_check(rd, RRDDIM_FLAG_DONT_DETECT_RESETS_OR_OVERFLOWS)?"noreset":""
+    );
+
+    metalog_commit_creation_record(ctx, buffer);
+}
+
+void metalog_commit_delete_dimension(RRDDIM *rd)
+{
+    struct metalog_instance *ctx;
+    BUFFER *buffer;
+    RRDSET *st = rd->rrdset;
+    RRDHOST *host = st->rrdhost;
+    char uuid_str[37];
+
+    /* Metadata are only available with dbengine */
+    if (!host->rrdeng_ctx || RRD_MEMORY_MODE_DBENGINE != st->rrd_memory_mode)
+        return;
+
+    ctx = host->rrdeng_ctx->metalog_ctx;
+    buffer = buffer_create(64); /* This will be freed after it has been committed to the metadata log buffer */
+
+    buffer_sprintf(buffer, "CONTEXT %s\n", st->id);
+
+    uuid_unparse_lower(*rd->state->rrdeng_uuid, uuid_str);
+    buffer_sprintf(buffer, "TOMBSTONE %s\n", uuid_str);
+
+    metalog_commit_creation_record(ctx, buffer);
+}
+
+/*
+ * Returns 0 on success, negative on error
+ */
+int metalog_init(struct rrdengine_instance *rrdeng_parent_ctx)
+{
+    struct metalog_instance *ctx;
+    int error;
+
+    rrdeng_parent_ctx->metalog_ctx = ctx = callocz(1, sizeof(*ctx));
+
+    ctx->max_disk_space = 32 * 1048576LLU;
```

Comment:
This should be configurable

Reply:
We will implement this during compaction.